### PR TITLE
Fix Filter on custom field (selection) can't be changed once filter is configured

### PR DESCRIPTION
**Pull request by timizuoebideri1**
# Closes: #3196 
# What's Changed

- Because the custom field(selection) on the Default filter form and the DynamicFilterForm had different field types, the DynamicFilterForm field did not update when the choice field on the Default filter form did. The correct field type for DefaultFilter is a choice field, whereas the field type for DynaicFilterForm is a multichar field.

- Previously, this updating of fields on DynamicFilterForm and DefualtFilterForm was only performed on select fields. Added JavaScript logic to update on input fields also.

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/core/tests/integration/test_filters.py [hunk 1]
```diff
@@ -15,6 +18,20 @@ def setUp(self):
         self.login(self.user.username, self.password)
         factory.RegionFactory.create_batch(15, has_parent=False)
         factory.SiteFactory.create_batch(15, has_tenant=False)
+        # set test user to admin
+        self.user.is_superuser = True
+        self.user.save()
+
+        self.cf_text_field_name = "text_field"
+        self.cf_integer_field_name = "integer_field"
+        self.cf_select_field_name = "select_field"
```
Comment (glennmatthews) [inline nautobot/core/tests/integration/test_filters.py:27]: Mind adding a # TODO comment here about changing these from `name` to `slug` in #824?

File: changes/3335.fixed [hunk 2]
```diff
@@ -0,0 +1 @@
+Fix Filter on custom field (selection) can't be changed once filter is configured.
```
Comment (glennmatthews) [inline changes/3335.fixed:1]:
```suggestion
Fixed inability to change filtering on custom field (selection) once filter is configured.
```

**Review by glennmatthews — COMMENTED** [review]
Haven't tested it myself but it looks reasonable. :-)

File: nautobot/utilities/utils.py [hunk 3]
```diff
@@ -869,7 +869,9 @@ def get_filterset_parameter_form_field(model, parameter):
     form_field = field.field
 
     # TODO(Culver): We are having to replace some widgets here because multivalue_field_factory that generates these isn't smart enough
-    if isinstance(field, NumberFilter):
+    if isinstance(field, CustomFieldFilterMixin):
+        form_field = field.custom_field.to_form_field()
```
Comment (bryanculver) [inline nautobot/utilities/utils.py:873]: Sadly long-term, we should merge this logic here (in `get_filterset_parameter_form_field`) with `multivalue_field_factory` and the field+filter+widget association.

**Review by bryanculver — APPROVED** [review]
Validated locally as well.

**Pull request merged**